docs/modules/ROOT/pages/documentation/extensions/grpc/README.adoc

```diff
@@ -0,0 +1,401 @@
+
+Example:
+[source, shell]
+----
+${webapp root folder} > mvn clean install
+----
+
+The stubs should be generated with similar structure like the following:
+
+* target
+** generated-sources
+*** protobuf
+**** grpc-java
+***** fish.payara.samples.grpc (*here you can see the service class*)
+**** java
+***** fish.payara.samples.grpc (*here you can see the types associated to the service*)
+
+Then you must create a class to extend the internal generated class with ImplBase particle. In our case, this class is located into PayaraServiceGrpc under the folder `/target/generated-sources/protobuf/grpc-java` .
+
+[[service-code-implementation]]
+==== Service Code Implementation
+
+[source, Java]
+----
+@Dependent
+public class PayaraService extends PayaraServiceGrpc.PayaraServiceImplBase {
+    private final static Logger log = Logger.getLogger(PayaraService.class.getName());
+    @Override
+    public void communicate(fish.payara.samples.grpc.PayaraReq request,
+                            io.grpc.stub.StreamObserver<fish.payara.samples.grpc.PayaraResp> responseObserver) {
+        final String message = request.getMessage();
+        log.info(String.format("Processing message: %s", message));
+        responseObserver.onNext(response(message));
+        responseObserver.onCompleted();
+    }
+
+    private static final fish.payara.samples.grpc.PayaraResp response(String message) {
+        return fish.payara.samples.grpc.PayaraResp.newBuilder() 
+                .setMessage(message) 
+                .build();
+    }
+}
+----
+
+Our example is a simple echo service that will print the HelloWorld message. On the next section when creating the client you will see more details.
+
+Finall step is to deploy the application to the server. 
+
+[[excluding-dependencies]]
+==== Excluding dependencies
+
+IMPORTANT: Other restriction is to not include any of the google dependencies to the lib folder of the war file. To achieve this you can use the the `maven-war-plugin` on the build section of the pom.xml file.
+
+Example: 
+[source,XML]
+----
+    <plugin>
+        <artifactId>maven-war-plugin</artifactId>
+        <version>2.4</version>
+        <configuration>
+            <packagingExcludes>
+                WEB-INF/lib/*.jar
+            </packagingExcludes>
+        </configuration>
+    </plugin>
+----
+
+For this example all the dependencies were excluded because all of them are related to io.grpc and protobuf artifacts that came from google. Those are provided from the jar file added to the modules folder. By doing this we are preventing issues during the deployment process.
+
+[[how-to-create-a-client]]
+== How to create a client?
+
+After deploying the server application now we can create a client. To achieve this it is needed to include stub generated files to the client application. You can create a new maven application and include the stub files on a package within the application. 
```

Comment:
Might be better to also list the necessary components to make the client application work in a clear manner.